src/openms/include/OpenMS/METADATA/SpectrumMetaDataLookup.h

```diff
@@ -264,16 +280,30 @@ namespace OpenMS
        @return True if all peptide IDs could be annotated successfully (including if all already had RT values), false otherwise.
 
        Look-up works by matching the "spectrum_reference" (meta value) of a peptide ID to the native ID of a spectrum. Only peptide IDs without RT (where PeptideIdentification::getRT() returns "NaN") are looked up; the RT is set to that of the corresponding spectrum.
-
-       The raw data is only loaded from @p filename if necessary, i.e. if there are any peptide IDs with missing RTs.
     */
-    static bool addMissingRTsToPeptideIDs(
-      std::vector<PeptideIdentification>& peptides, const String& filename,
+    static bool addMissingRTsToPeptideIDs(std::vector<PeptideIdentification>& peptides, const String &filename,
       bool stop_on_error = false);
 
+    /**
+     * @brief Add missing "spectrum_reference"s to peptide identifications based on raw data
+     *
+     * @param peptides Peptide IDs with or without spectrum_reference
+     * @param lookup SpectrumMetaDataLookup with loaded spectra, metadata and optional spectra_data value for Proteins
+     * @param stop_on_error Stop when an ID could not be matched to a spectrum (or keep going)?
+     * @param override_spectra_data if given ProteinIdentifications should be updated with new "spectra_data" values from SpectrumMetaDataLookup
+     * @param proteins
+     *
+     * @return True if all peptide IDs could be annotated successfully (including if all already had "spectrum_reference" values), false otherwise.
+     *
+     * Look-up works by matching RT of a peptide ID to the native ID of a spectrum. All spectrum_references are updated/added.
```

Comment:
why are RTs of an identification matched to the native ID of a spectrum? is matching done between RTs? not clear how it works from the comment. or is RT wrong? also do peptide ID and native ID mean something else here? Identification and Identifier? if so it might be confusing (applies to other places in comments as well)